sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -68,9 +68,18 @@ trait CheckAnalysis extends PredicateHelper {
       case e if e.dataType != IntegerType => failAnalysis(
         s"The limit expression must be integer type, but got " +
           e.dataType.catalogString)
-      case e if e.eval().asInstanceOf[Int] < 0 => failAnalysis(
```

Comment:
How about this:

```scala
case e => e.eval() match {
  case null => failAnalysis(
    s"The evaluated limit expression must not be null, but got ${limitExpr.sql}")
  case v: Int if v < 0 => failAnalysis(
    s"The limit expression must be equal to or greater than 0, but got $v")
  case _ => // OK
}

```

Reply:
I'm personally not a fan of this because of the nesting, but I guess it's a matter of taste. Is there any guidelines for this regarding to styling? I don't have a strong opinion about this

Reply:
nope but it's shorter and I don't think this code makes less readable.

Reply:
fixed